flutter_local_notifications_linux/lib/flutter_local_notifications_linux.dart

```diff
@@ -1,7 +1,12 @@
 /// The Linux implementation of `flutter_local_notifications`.
 library flutter_local_notifications_linux;
 
-export 'src/flutter_local_notifications.dart';
+// flutter_local_notifications_linux depends on dbus and posix
+// which uses FFI internally; export a stub for platforms that don't
+// support FFI (e.g., web) to avoid having transitive dependencies
+// break web compilation.
+export 'src/flutter_local_notifications_stub.dart'
+    if (dart.library.ffi) 'src/flutter_local_notifications.dart';
```

Comment:
I know this logic is based on what's in the community plus plugins but given the comments and intent, perhaps the logic could be better expressed. The code reads like it should load the real implementation for platform thats support FFI but the intent is to load a stub for the web so apps can compile when targeting web. This could be done by kind of inverting the logic

```
export 'src/flutter_local_notifications.dart'
     if (dart.library.js) 'src/flutter_local_notifications_stub.dart';
```

reference: https://github.com/flutter/cocoon/blob/eb8d951bf547256785229538efc47d3459906155/app_flutter/lib/service/downloader.dart#L5

Reply:
Hmm actually ignore my comment. From a generic POV, you'd want to see if the APIs you use are actually supported

Reply:
It really doesn't make much of a difference in the case of the Web: it will work in both cases. But if the Web ever gets FFI support, or if there is another platform that doesn't support FFI, my export option will make more sense, because it's not just limited to the Web, but any platform that doesn't support FFI.
I can change this now if you wish.